### ISSUE TITLE
Missing labels for donation form in fundraising page

### ISSUE THREAD

**Issue by erosselli** [django/djangoproject.com#2534]
The [fundraising page](https://www.djangoproject.com/fundraising/) has a donation form (under the "Help us make it happen" section) that has no accesible names for the inputs. 

<img width="389" height="237" alt="donation form with no accessible labels" src="https://github.com/user-attachments/assets/40ef42ae-cd1b-4c6b-83d4-d132d131d633" />

The input components in the form should have `<label>` elements.

Team Mars will be working on this as part of Djangonaut Space Session 6.

**Comment by Skyiesac** [comment]
Hey Eli, can you assign this to me ?
